extension/js/common/platform/xss.ts

```diff
@@ -98,16 +98,20 @@ export class Xss {
         } else if (!src) {
           img.remove(); // src that exists but is null is suspicious
         } else if (imgHandling === 'IMG-TO-LINK') { // replace images with a link that points to that image
-          const title = img.getAttribute('title');
-          img.removeAttribute('src');
-          const a = document.createElement('a');
-          a.href = src;
-          a.className = 'image_src_link';
-          a.target = '_blank';
-          a.innerText = (title || 'show image') + (src.startsWith('data:image/') ? '' : ' (remote)');
-          const heightWidth = `height: ${img.clientHeight ? `${Number(img.clientHeight)}px` : 'auto'}; width: ${img.clientWidth ? `${Number(img.clientWidth)}px` : 'auto'};max-width:98%;`;
-          a.setAttribute('style', `text-decoration: none; background: #FAFAFA; padding: 4px; border: 1px dotted #CACACA; display: inline-block; ${heightWidth}`);
-          Xss.replaceElementDANGEROUSLY(img, a.outerHTML); // xss-safe-value - "a" was build using dom node api
+          if (src.startsWith('data:image/')) {
+            const title = img.getAttribute('title');
+            img.removeAttribute('src');
+            const a = document.createElement('a');
+            a.href = src;
+            a.className = 'image_src_link';
+            a.target = '_blank';
+            a.innerText = title || 'show image';
+            const heightWidth = `height: ${img.clientHeight ? `${Number(img.clientHeight)}px` : 'auto'}; width: ${img.clientWidth ? `${Number(img.clientWidth)}px` : 'auto'};max-width:98%;`;
+            a.setAttribute('style', `text-decoration: none; background: #FAFAFA; padding: 4px; border: 1px dotted #CACACA; display: inline-block; ${heightWidth}`);
+            Xss.replaceElementDANGEROUSLY(img, a.outerHTML); // xss-safe-value - "a" was build using dom node api
+          } else {
```

Comment:
Do I understand correctly, that `https://*.googleusercontent.com/...` images are now blocked too?

Reply:
Any remote images are blocked now.
Maybe you want to whitelist some domains?

Reply:
> Any remote images are blocked now. Maybe you want to whitelist some domains?

If we don't have any use cases where such a link appears naturally by pasting/dragging files to the secure compose box, then we're good to go.